src/consensus.rs

```diff
@@ -313,14 +312,20 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                         if existing_term != leader_prev_log_term {
                             scoped_debug!("AppendEntriesRequest: inconsistent previous log term: leader: {}, local: {}",
                                           leader_prev_log_term, existing_term);
-                            messages::append_entries_response_inconsistent_prev_entry(self.current_term())
+                            // If an existing entry conflicts with a new one (same index but different terms),
+                            // delete the existing entry and all that follow it
+                            messages::append_entries_response_inconsistent_prev_entry(self.current_term(),
+                                leader_prev_log_index)
```

Comment:
Ah, I see. Maybe update the comment so that it reflects that we're telling the leader to overwrite our log beginning after `leader_prev_log_index`.